pkg/pquerier/select.go

```diff
@@ -349,19 +376,30 @@ func (queryCtx *selectQueryContext) processQueryResults(query *partQuery) error
 				results.IsServerAggregates(),
 				queryCtx.showAggregateLabel)
 			if err != nil {
-				return err
+				queryCtx.errorChannel <- err
+				queryCtx.createDFLock.Unlock()
```

Comment:
Are you sure writing to `queryCtx.errorChannel <- err` can't block, potentially deadlocking `createDFLock`?